network/network.go

```diff
@@ -172,20 +188,38 @@ func (n *Network) Config() interface{} {
 // Start initiates the Network subsystem
 func (n *Network) Start() error {
 	n.startTime.Store(time.Now())
+
+	// Load DAG and start publishing
 	n.publisher.Subscribe(dag.AnyPayloadType, n.lastTransactionTracker.process)
 	n.publisher.Start()
-
 	if err := n.graph.Verify(context.Background()); err != nil {
 		return err
 	}
 
+	// Sanity check for configured node DID: can we resolve it?
+	if n.configuredNodeDID != nil {
+		doc, _, err := n.didDocumentResolver.Resolve(*n.configuredNodeDID, nil)
+		if err != nil {
+			return fmt.Errorf("invalid NodeDID configuration: DID document can't be resolved (did=%s): %w", n.configuredNodeDID, err)
+		}
+		if len(doc.KeyAgreement) == 0 {
+			return fmt.Errorf("invalid NodeDID configuration: DID document does not contain a keyAgreement key (did=%s)", n.configuredNodeDID)
+		}
+		kid := doc.KeyAgreement[0].ID.String()
```

Comment:
I'm wondering. Does it *have* to be the first `KeyAgreement` or can it also be the second one for example?

Reply:
It now checks all.